crates/typst-syntax/src/source.rs

```diff
@@ -35,12 +35,15 @@ impl Source {
     pub fn new(id: FileId, text: String) -> Self {
         let mut root = parse(&text);
         root.numberize(id, Span::FULL).unwrap();
-        Self(Arc::new(Repr {
-            id,
-            lines: lines(&text),
-            text: Prehashed::new(text),
-            root: Prehashed::new(root),
-        }))
+        Self(
+            Repr {
+                id,
+                lines: lines(&text),
+                text: text.into(),
+                root: root.into(),
```

Comment:
While this isn't as bad as the `Arc` case, I'm personally in favor of keeping the explicit `Prehashed::new`.